src/ng-xform/datepicker-field/datepicker-field.component.ts

```diff
@@ -0,0 +1,76 @@
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Component, AfterContentInit, ViewChild, ElementRef, Renderer, AfterViewInit } from '@angular/core';
+import { BsDatepickerConfig, BsLocaleService } from 'ngx-bootstrap/datepicker';
+
+import { BaseDynamicFieldComponent } from '../field-components/base-dynamic-field.component';
+import { DatepickerField } from '../fields';
+
+/**
+ * Component to generate a bootstrap form field of general type
+ *
+ * :editing: Flag to control component state
+ * :form: FormGroup containing the field
+ * :field: Intance of field configurations
+ */
+@Component({
+  selector: 'ng-xform-datepicker-field',
+  templateUrl: './datepicker-field.component.html',
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: DatepickerFieldComponent,
+    multi: true
+  }],
+})
+export class DatepickerFieldComponent extends BaseDynamicFieldComponent<DatepickerField> implements AfterViewInit, AfterContentInit,
+  ControlValueAccessor {
+  config: Partial<BsDatepickerConfig>;
+  componentControl = new FormControl();
+  private input: HTMLInputElement;
+
+  _onChange = (value: any) => { };
+  _onTouched = () => { };
+
+  constructor(private _localeService: BsLocaleService, private elementRef: ElementRef, private renderer: Renderer) {
```

Comment:
Explain (in the docs):
- Why are you injecting BsLocaleService?
- Why are you injecting elementRef?

Reply:
OK